hbase-backup/src/test/java/org/apache/hadoop/hbase/backup/TestIncrementalBackup.java

```diff
@@ -224,7 +241,92 @@ public void TestIncBackupRestore() throws Exception {
       hTable = conn.getTable(table2_restore);
       Assert.assertEquals(NB_ROWS_IN_BATCH + 5, HBaseTestingUtil.countRows(hTable));
       hTable.close();
+
+      // Test bulkloads
+      newTable1Desc = TableDescriptorBuilder.newBuilder(newTable1Desc)
+        .setColumnFamily(ColumnFamilyDescriptorBuilder.of(fam3Name)).build();
+      updateDesc(newTable1Desc, tables, conn);
+      testBulkloads(conn, famName, mobName);
       admin.close();
     }
   }
+
+  private void updateDesc(TableDescriptor desc, List<TableName> tablesToBackup, Connection conn)
+    throws Exception {
+    TEST_UTIL.getAdmin().modifyTable(desc);
+    BackupAdminImpl backupAdmin = new BackupAdminImpl(conn);
+    BackupRequest req = createBackupRequest(BackupType.FULL, tablesToBackup, BACKUP_ROOT_DIR);
+    String backupId = backupAdmin.backupTables(req);
+    checkSucceeded(backupId);
+  }
+
+  private void testBulkloads(Connection conn, byte[]... famsToLoad) throws Exception {
+    BackupAdminImpl backupAdmin = new BackupAdminImpl(conn);
+    List<HRegion> allRegions = TEST_UTIL.getHBaseCluster().getRegions(table1);
+    HRegion regionToBulkload = allRegions.get(0);
+    String regionName = regionToBulkload.getRegionInfo().getEncodedName();
+    Path regionDir = createHFiles(table1, regionName, famsToLoad);
+    doBulkload(table1, regionDir);
+    TEST_UTIL.waitTableAvailable(table1);
+
+    BackupRequest request =
+      createBackupRequest(BackupType.INCREMENTAL, Lists.newArrayList(table1), BACKUP_ROOT_DIR);
+    String incrementalBackupId = backupAdmin.backupTables(request);
+    assertTrue(checkSucceeded(incrementalBackupId));
+    TableName[] fromTables = new TableName[] { table1 };
+    TableName[] toTables = new TableName[] { table1_restore };
+
+    backupAdmin.restore(BackupUtils.createRestoreRequest(BACKUP_ROOT_DIR, incrementalBackupId,
+      false, fromTables, toTables, true));
+
+    int rowsExpected = HBaseTestingUtil.countRows(conn.getTable(table1), famsToLoad);
+
+    Table hTable = conn.getTable(table1_restore);
+    Assert.assertEquals(HBaseTestingUtil.countRows(hTable, famsToLoad), rowsExpected);
+  }
+
+  private static Path createHFiles(TableName tn, String regionName, byte[]... fams)
```

Comment:
I'd actually expect that the `TestFullBackup` also needs to test bulkloads. So either a similar method to generate HFiles already exists, or this method can move to a higher level so it can be used in multiple tests.